bootstrap/090_post_setup.sql

```diff
@@ -70,21 +70,84 @@ CREATE OR REPLACE TASK TASKS.WAREHOUSE_EVENTS_MAINTENANCE
+    task_run_id text default (select INTERNAL.TASK_RUN_ID());
+BEGIN
+    let query_id text := (select query_id from table(information_schema.task_history(TASK_NAME => :task_name)) WHERE GRAPH_RUN_GROUP_ID = :task_run_id  AND DATABASE_NAME = current_database() limit 1);
+    let simple_tables resultset := (SELECT t.table_name, t.index_col FROM (VALUES
+        ('SERVERLESS_TASK_HISTORY', 'end_time'),
+        ('TASK_HISTORY', 'completed_time'),
+        ('SESSIONS', 'created_on'),
+        ('WAREHOUSE_METERING_HISTORY', 'end_time'),
+        ('LOGIN_HISTORY', 'event_timestamp'),
+        ('HYBRID_TABLE_USAGE_HISTORY', 'end_time'),
+        ('MATERIALIZED_VIEW_REFRESH_HISTORY', 'end_time')
+    ) AS t(table_name, index_col));
+    let cur cursor for simple_tables;
+    FOR rowvar IN cur DO
+        let table_name text := rowvar.table_name;
+        let index_col text := rowvar.index_col;
+        BEGIN
+            let start_time timestamp_ltz := (select current_timestamp());
+            let input object := (select output from INTERNAL.TASK_LOG where success and task_name = :task_name and object_name = :table_name order by task_start desc limit 1);
+            INSERT INTO INTERNAL.TASK_LOG(task_start, task_run_id, query_id, input, task_name, object_name) select :start_time, :task_run_id, :query_id, :input, :task_name, :table_name;
+
+            let output variant;
+            CALL INTERNAL.refresh_simple_table(:table_name, :index_col, true, :input) into :output;
+
+            let success boolean := (select :output['SQLERRM'] is null);
+            UPDATE INTERNAL.TASK_LOG SET success = :success, output = :output, task_finish = current_timestamp() WHERE task_start = :start_time AND task_run_id = :task_run_id and task_name = :task_name AND object_name = :table_name;
+        END;
+    END FOR;
+
+    -- TODO add table log
+    call internal.refresh_warehouses();
+END;
 
 CREATE OR REPLACE TASK TASKS.QUERY_HISTORY_MAINTENANCE
     SCHEDULE = '60 minute'
     ALLOW_OVERLAPPING_EXECUTION = FALSE
     USER_TASK_MANAGED_INITIAL_WAREHOUSE_SIZE = "LARGE"
     AS
-    CALL INTERNAL.refresh_queries(true);
+DECLARE
+    start_time timestamp_ltz default (select current_timestamp());
+    task_name text default 'QUERY_HISTORY_MAINTENANCE';
+    object_name text default 'QUERY_HISTORY';
+    task_run_id text default (select INTERNAL.TASK_RUN_ID());
+BEGIN
+    let query_id text := (select query_id from table(information_schema.task_history(TASK_NAME => :task_name)) WHERE GRAPH_RUN_GROUP_ID = :task_run_id  AND DATABASE_NAME = current_database() limit 1);
+    let input variant := (select output from INTERNAL.TASK_LOG where success AND task_name = :task_name AND object_name = :object_name order by task_start desc limit 1);
+    INSERT INTO INTERNAL.TASK_LOG(task_start, task_run_id, query_id, input, task_name, object_name) select :start_time, :task_run_id, :query_id, :input, :task_name, :object_name;
```

Comment:
as I look at all these inserts/updates... maybe we should make two procedures that encapsulate these behaviors?